etcdserver/server.go

```diff
@@ -632,6 +628,21 @@ func (s *EtcdServer) Lead() uint64 { return atomic.LoadUint64(&s.r.lead) }
 
 func (s *EtcdServer) Leader() types.ID { return types.ID(s.Lead()) }
 
+// RequestTimeout returns an appropriate timeout to wait for request to be committed.
+func (s *EtcdServer) RequestTimeout() time.Duration {
```

Comment:
I have some concern for max timeout. The max timeout could be as long as 2min if it takes care of leader switch, which could take 2*maxElectionTimeout. If people use etcd to do health check for 10s, one proposal lost could make it failed.

Maybe i am over-concerning. Any thoughts?